bindings/jupyroot/CMakeLists.txt

```diff
@@ -41,13 +41,10 @@ foreach(val RANGE ${how_many_pythons})
 
   set(libname JupyROOT${python_under_version_string})
 
-  add_library(${libname} SHARED src/IOHandler.cxx)
-
   # libJupyROOT uses ROOT headers from source dirs and depends on Core
+  ROOT_LINKER_LIBRARY(${libname} src/IOHandler.cxx DEPENDENCIES ${python_library} Core ${CMAKE_DL_LIBS} CMAKENOEXPORT)
```

Comment:
@oshadura what do you think of using the ROOT macro instead of just the CMake standard one?

Motivation to use the ROOT macro: the ROOT macro already controls that the library is generated with the `.so` suffix. If we want to do the same with the CMake macros, we can, but then we need to add an extra `set(CMAKE_SHARED_LIBRARY_SUFFIX ".so")`.

Reply:
@maxgalli do you know what `${CMAKE_DL_LIBS}` is and why is needed here? I know it was there in the CMake macro, but I don't see it e.g. in `libROOTPythonizations` which is created with `ROOT_LINKER_LIBRARY` too.

Reply:
it's this
https://cmake.org/cmake/help/latest/variable/CMAKE_DL_LIBS.html
but honestly I don't know why it's needed here and not in other places

Reply:
I believe it could have been needed in the past, when we used `dlopen` from the code of JupyROOT, now we just import the library as a regular Python module (since it is an extension module). I would try removing `CMAKE_DL_LIBS` and check the builds.